cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDbTemplate.java

```diff
@@ -463,39 +474,56 @@ public void delete(final Item item, final PersistenceExceptionHandler<?>... pers
         final List<T> totalItems = new ArrayList<>();
         Map<String, AttributeValue> lastEvaluatedKey = null;
         final String tableName = databaseSchemaHolder.schemaName() + "." + itemConfiguration.tableName();
+
         if (itemConfiguration.hasIndexOn(query.getAttributeName())) {
             do {
                 final String queryAttributeName = query.getAttributeName();
                 final PrimaryKeyDefinition primaryKeyDefinition = itemConfiguration.primaryKeyDefinition();
                 final String primaryKeyPropertyName = primaryKeyDefinition.propertyName();
                 final boolean isPrimaryKeyQuery = queryAttributeName.equals(primaryKeyPropertyName);
+
                 final QueryRequest queryRequest = new QueryRequest().withTableName(tableName)
                         .withKeyConditions(conditions).withExclusiveStartKey(lastEvaluatedKey);
+
+                if (maxPageSizeOverridden(maxPageSize)) {
```

Comment:
`maxPageSize > 0` would be clearer. Perhaps `limit` is better than `maxPageSize`?